x/mongo/driver/topology/rtt_monitor_test.go

```diff
@@ -219,6 +222,106 @@ func TestRTTMonitor(t *testing.T) {
 			rtt.reset()
 		}
 	})
+
+	// GODRIVER-2464
+	// Test that the RTT monitor can continue monitoring server RTTs after an operation gets stuck.
+	// An operation can get stuck if the server or a proxy stops responding to requests on the RTT
+	// connection but does not close the TCP socket, effectively creating an operation that will
+	// never complete.
+	t.Run("stuck operations time out", func(t *testing.T) {
+		t.Parallel()
+
+		// Start a goroutine that listens for and accepts TCP connections, reads requests, and
+		// responds with {"ok": 1}. The first 2 connections simulate "stuck" connections and never
+		// respond or close.
+		l, err := net.Listen("tcp", "localhost:0")
+		require.NoError(t, err)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for i := 0; ; i++ {
+				conn, err := l.Accept()
+				if err != nil {
+					// The listen loop is cancelled by closing the listener, so there will always be
+					// an error here. Log the error to make debugging easier in case of unexpected
+					// errors.
+					t.Logf("Accept error: %v", err)
+					return
+				}
+
+				// Only close connections when the listener loop returns to prevent closing "stuck"
+				// connections while the test is running.
+				defer conn.Close()
+
+				wg.Add(1)
+				go func(i int) {
```

Comment:
Why does the reading and writing from and to the connection need to happen within yet another goroutine?

Reply:
If we try to handle a monitoring connection (i.e. read "hello" requests and write "hello" responses) in the same goroutine, we wouldn't be able to accept any new connections until the current one closes. While that's not a requirement of this test, listening for and handling each connection are logically concurrent tasks, so running them in separate goroutines makes the behavior of the mock server easier to understand and more like an actual server.